plugins/modules/remote_management/ipmi/ipmi_boot.py

```diff
@@ -138,6 +144,7 @@ def main():
             port=dict(default=623, type='int'),
             user=dict(required=True, no_log=True),
             password=dict(required=True, no_log=True),
+            key=dict(required=False, type='str', no_log=True),
```

Comment:
`required=false` is redundant:
```suggestion
            key=dict(type='str', no_log=True),
```

Reply:
Thank you for the feedback! I took this from another module - these `ipmi_*` modules have a bit different style.

The other "string" args do not specify `type='str'` so I assume that is the default. While I am at it, should I update all the string args to explicitly have `type='str'`?

Reply:
Well, this (even the required thing) is more a "good practice" than a 100% strict rule. WRT required we had a general consensus that it was better to go without "required=False". I don't recall having reached a conclusion about type=str, so it boils down to personal preferences. Personally, I would prefer being explicit about that but will not push it.

Reply:
Thank you, I will take this into account for the next version.

Reply:
This is done also.